### PR TITLE
Encrypt master password by default at sign in

### PR DESCRIPTION
**Pull request by kir0ul**
This PR closes https://github.com/lesspass/lesspass/issues/597

Just for the record, this is how the sign in looks like with this PR:

image

### CODE REVIEW

File: packages/lesspass-pure/src/i18n/en.json [hunk 1]
```diff
@@ -51,7 +51,7 @@
   "SiteLoginMasterPasswordMandatory": "Site, login, and master password fields are mandatory.",
   "SorryCopy": "Sorry, copying only works in modern browsers.",
   "UpdateYourSearch": "Please try broadening your search.",
-  "Username": "Username",
```
Comment (guillaumevincent) [inline packages/lesspass-pure/src/i18n/en.json:54]: Don't remove this key, it's used on other pages

File: packages/lesspass-pure/src/i18n/fr.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
   "SiteLoginMasterPasswordMandatory": "Les champs site, login et mot de passe fort sont obligatoires.",
   "SorryCopy": "Nous sommes désolés, la copie ne fonctionne que sur les navigateurs modernes",
   "UpdateYourSearch": "Merci de modifier votre recherche.",
-  "Username": "Nom d'utilisateur",
```
Comment (guillaumevincent) [inline packages/lesspass-pure/src/i18n/fr.json:54]: Same here

File: packages/lesspass-pure/src/views/Login.vue [hunk 3]
```diff
@@ -37,6 +37,7 @@
         v-bind:email="email"
         v-bind:showEncryptButton="true"
         v-bind:EncryptButtonText="$t('Encrypt my master password')"
+        v-bind:PlainCheckboxText="$t('Use plain password')"
```
Comment (guillaumevincent) [inline packages/lesspass-pure/src/views/Login.vue:40]: Master password component is used in other pages.

I'm not sure it's interesting to have this as a props.
You can use `$t('Use plain password')` directly inside the component.

or you change everywhere `<master-password/>` component (I prefer the first solution)

**Comment by guillaumevincent** [comment]
Thank you `kir0ul`, some small changes before I can merge this

File: packages/lesspass-pure/src/i18n/en.json (hunk 1 above)
Comment (kir0ul) [reply packages/lesspass-pure/src/i18n/en.json:54]: Oops good catch, I guess I screwed up with the conflicts I had when I rebased on `master` :grimacing:

File: packages/lesspass-pure/src/i18n/fr.json (hunk 2 above)
Comment (kir0ul) [reply packages/lesspass-pure/src/i18n/fr.json:54]: Fixed!

File: packages/lesspass-pure/src/views/Login.vue (hunk 3 above)
Comment (kir0ul) [reply packages/lesspass-pure/src/views/Login.vue]: Should be fixed!

File: packages/lesspass-pure/src/views/Login.vue [hunk 4]
```diff
@@ -105,27 +117,31 @@ export default {
...
-                  "LoginIncorrectError",
-                  "The email and password you entered did not match our records. Please double-check and try again."
-                )
-              );
-            } else {
-              message.displayGenericError();
-            }
+        encryptPassword(this.email, this.password)
+          .then(encryptedPassword => {
+            let pass = document.getElementById("checkPlainPassword").checked ? this.password : encryptedPassword;
```
Comment (kir0ul) [inline packages/lesspass-pure/src/views/Login.vue:122]: Beginner question: is this the best way to test if the checkbox is checked or is there a better way, like using a prop or something else?

File: packages/lesspass-pure/src/i18n/en.json [hunk 5]
```diff
@@ -53,6 +53,7 @@
   "SorryCopy": "Sorry, copying only works in modern browsers.",
   "UpdateYourSearch": "Please try broadening your search.",
   "Username": "Username",
+  "Use plain password": "Use plain password (not recommended)",
```
Comment (kir0ul) [inline packages/lesspass-pure/src/i18n/en.json:56]: I'm not sure why I don't get the **full** translation string of the _Use plain password_ checkbox in English. If I switch my browser in French I get it but not in English. Any hints?

File: packages/lesspass-pure/src/components/MasterPassword.vue [hunk 6]
```diff
@@ -33,26 +33,44 @@
...
-      type="button"
-      class="btn btn-link btn-sm p-0"
-      v-if="showEncryptButton"
-      v-on:click="encryptMasterPassword()"
-      v-bind:class="{'disabled': email === ''}"
-    >
-      <small>{{ EncryptButtonText }}</small>
-    </button>
+    <small>
+      <div class="form-check form-switch" v-if="showEncryptButton">
```
Comment (kir0ul) [inline packages/lesspass-pure/src/components/MasterPassword.vue:63]: Just for the record, this widget will be a [switch in Bootstrap 5](https://getbootstrap.com/docs/5.0/forms/checks-radios/#switches), but for now there's no switch widget in Bootstrap 4 :cry:

File: packages/lesspass-pure/src/views/Login.vue (hunk 4 above)
Comment (guillaumevincent) [reply packages/lesspass-pure/src/views/Login.vue:122]: `v-model="checkPlainPassword"` on the html element then `this.checkPlainPassword` in the vue function should be better 
See https://vuejs.org/v2/guide/forms.html

File: packages/lesspass-pure/src/i18n/en.json (hunk 5 above)
Comment (guillaumevincent) [reply packages/lesspass-pure/src/i18n/en.json:56]: Hum, Maybe it use `$t("Use plain password")` directly. Ignore the translation in this PR, I will look at it at the end.
I will run automatic scripts to do it for us.

File: packages/lesspass-pure/src/components/MasterPassword.vue (hunk 6 above)
Comment (guillaumevincent) [reply packages/lesspass-pure/src/components/MasterPassword.vue:63]: Checkbox is consistent with the main view. I will see if we can do the bootstrap 5 migration easily without breaking everything.

File: packages/lesspass-pure/src/i18n/en.json (hunk 5 above)
Comment (kir0ul) [reply packages/lesspass-pure/src/i18n/en.json:56]: Sounds good!

File: packages/lesspass-pure/src/views/Login.vue (hunk 4 above)
Comment (kir0ul) [reply packages/lesspass-pure/src/views/Login.vue]: Tried to modify this in https://github.com/lesspass/lesspass/pull/604/commits/fa145da93e30b8aaaed654304ca06069140cc612 according to your suggestion.

**Comment by kir0ul** [comment]
Should I do anything more about testing? I feel the checkbox is not really tested at the moment, but not sure where to plug some test.

**Comment by guillaumevincent** [comment]
No it's ok, I will try to merge this this week
Thank you for your time, I hope you enjoy the process

**Comment by kir0ul** [comment]
I did! Can't wait to see it in my browser! :slightly_smiling_face:

**Comment by guillaumevincent** [comment]
I fixed the implementation to encrypt the default password in the other views.
I will do more tests early next week before I go to production.
Thank you again @kir0ul

**Pull request merged**